client/tests/testgeometry/test_utils.py

```diff
@@ -28,3 +30,34 @@ def test_incircle(self):
         self.assertLess(incircle(a, b, c, Vector(-11, 0)), 0)
         self.assertEqual(incircle(a, b, c, Vector(1, 1)), 0)
         self.assertEqual(incircle(a, b, c, Vector(1, -1)), 0)
+
+    def test_aabb(self):
+        a = Vector(2, 2)
+        b = Vector(6, 4)
+        rect = AABB(a, b)
+
+        self.assertEqual(rect.distance2(Vector(0, 0)), 8)
+        self.assertEqual(rect.distance2(a), 0)
+        self.assertEqual(rect.distance2(Vector(3, 3)), 0)
+        self.assertEqual(rect.distance2(Vector(4, 3)), 0)
+        self.assertEqual(rect.distance2(Vector(8, 5)), 5)
+        self.assertEqual(rect.distance2(Vector(1, -1)), 10)
+        self.assertEqual(rect.distance2(Vector(4, 1)), 1)
+        self.assertEqual(rect.distance2(Vector(9, 3)), 9)
+        self.assertEqual(rect.distance2(Vector(5, 5)), 1)
+        self.assertEqual(rect.distance2(Vector(0, 4)), 4)
+
+        self.assertEqual(rect.distance(Vector(0, 4)), 2)
+        self.assertEqual(rect.distance(Vector(9, 3)), 3.0)
+
+        self.assertEqual(rect.contains(a), True)
```

Comment:
Those are different tests, don't put everything in 1 test, split them into several. Look at vector tests